src/probnum/utils/randomutils.py

```diff
@@ -0,0 +1,20 @@
+from typing import Union
+
+import numpy as np
+
+
+def derive_random_seed(*args: Union[np.random.RandomState, np.random.Generator]) -> int:
```

Comment:
`combine_random_seeds` seems like a more descriptive name to me, similar to `boost::hash_combine`